osu.Game/Overlays/Comments/CommentsContainer.cs

```diff
@@ -13,17 +13,25 @@
 using System.Threading;
 using System.Linq;
 using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Graphics.Sprites;
 
 namespace osu.Game.Overlays.Comments
 {
     public class CommentsContainer : CompositeDrawable
     {
-        private readonly CommentableType type;
-        private readonly long id;
+        private CommentableType type;
+        private long id;
 
         public readonly Bindable<CommentsSortCriteria> Sort = new Bindable<CommentsSortCriteria>();
         public readonly BindableBool ShowDeleted = new BindableBool();
 
+        public void ShowComments(CommentableType type, long id)
+        {
+            this.type = type;
+            this.id = id;
+            Sort.TriggerChange();
```

Comment:
How about this being the method that clears and retrieves comments, and having the sort change event trigger `ShowComments(type, id)`?